### PR TITLE
[type:feat] complete h2 and mysql

### PR DESCRIPTION
**Pull request by aFlyBird0**
## Overview

* Both h2 and MySQL data sources are currently supported.
* Passed helm lint and test
* already tested in a real deployment.
* Chinese documentation is completed.

## Readme

**Note**:  you should use 

`helm repo add  shenyu https://aFlyBird0.github.io/incubator-shenyu-helm-chart` 

to add shenyu repo. Because the official repo hasn't been released now.

**README**:

https://github.com/apache/shenyu-helm-chart/pull/9/files#diff-12ca2065311179b039eb4ccf79e104ec50bd26e6814fa109b5a8aea90aaecf8b

## Chart Dry Run

```shell
helm repo add  bird-shenyu https://aFlyBird0.github.io/incubator-shenyu-helm-chart
helm repo update bird-shenyu
helm search repo bird-shenyu -l
```

### h2
**Command:** `helm install --debug --dry-run shenyu-dev ./charts/shenyu -n=shenyu`

Output:
```yaml
install.go:178: [debug] Original chart version: ""
install.go:195: [debug] CHART PATH: /Users/lhp/Code/Go/opensource/incubator-shenyu-helm-chart/charts/shenyu

NAME: shenyu-dev
LAST DEPLOYED: Wed Aug 17 19:53:35 2022
NAMESPACE: shenyu
STATUS: pending-install
REVISION: 1
TEST SUITE: None
USER-SUPPLIED VALUES:
{}

COMPUTED VALUES:
admin:
  enabled: true
  image: apache/shenyu-admin
  nodePort: 31095
bootstrap:
  enabled: true
  image: apache/shenyu-bootstrap
  nodePort: 31195
dataSource:
  active: h2
  h2:
    password: sa
    username: sa
  initEnabled: true
  mysql:
    driverClass: com.mysql.cj.Driver
    ip: null
    password: null
    port: 3306
    storageClass: null
    username: root
replicas: 1
version: 2.4.3

HOOKS:
MANIFEST:
---
# Source: shenyu/templates/ConfigMap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  namespace: shenyu
  name: shenyu-dev-configmap
  labels:
    "app.kubernetes.io/name": 'shenyu'
    "app.kubernetes.io/managed-by": "Helm"
    "app.kubernetes.io/component": "shenyu-configmap"
    "app.kubernetes.io/instance": "shenyu-dev"
data:
  application-local.yml: |
    server:
      port: 9195
      address: 0.0.0.0
    spring:
      profiles:
        active: h2
      main:
        allow-bean-definition-overriding: true
      application:
        name: shenyu-bootstrap
    management:
      health:
        defaults:
        enabled: false
    shenyu:
      local:
        enabled: true
      file:
        enabled: true
      cross:
        enabled: true
      dubbo:
        parameter: multi
      sync:
        websocket:
          urls: ws://shenyu-dev-admin.shenyu.svc.cluster.local:9095/websocket
      exclude:
        enabled: false
        paths:
        - /favicon.ico
      extPlugin:
        enabled: true
        threads: 1
        scheduleTime: 300
        scheduleDelay: 30
      scheduler:
        enabled: false
        type: fixed
        threads: 16
    logging:
      level:
        root: info
        org.springframework.boot: info
        org.apache.ibatis: info
        org.apache.shenyu.bonuspoint: info
        org.apache.shenyu.lottery: info
        org.apache.shenyu: info
  application-h2.yml: |
    shenyu:
      database:
        dialect: h2
        init_script: "sql-script/h2/schema.sql"
        init_enable: true
    spring:
      datasource:
        url: jdbc:h2:mem:~/shenyu;DB_CLOSE_DELAY=-1;MODE=MySQL;
        username: sa
        password: sa
        driver-class-name: org.h2.Driver
---
# Source: shenyu/templates/shenyu-admin-svc.yaml
# Now we only support NodePort mode
apiVersion: v1
kind: Service
metadata:
  namespace: shenyu
  name: shenyu-dev-admin
  labels:
    "app.kubernetes.io/name": 'shenyu-admin'
    "app.kubernetes.io/managed-by": "Helm"
    "app.kubernetes.io/component": "shenyu-admin"
    "app.kubernetes.io/instance": "shenyu-dev"
spec:
  selector:
    "app.kubernetes.io/component": "shenyu-admin"
    "app.kubernetes.io/instance": "shenyu-dev"
  type: NodePort
  ports:
    - protocol: TCP
      port: 9095
      targetPort: 9095
      nodePort: 31095
---
# Source: shenyu/templates/shenyu-bootstrap-svc.yaml
# Now we only support NodePort mode
apiVersion: v1
kind: Service
metadata:
  namespace: shenyu
  name: shenyu-dev-bootstrap
  labels:
    "app.kubernetes.io/name": 'shenyu-bootstrap'
    "app.kubernetes.io/managed-by": "Helm"
    "app.kubernetes.io/component": "shenyu-bootstrap"
    "app.kubernetes.io/instance": "shenyu-dev"
spec:
  selector:
    "app.kubernetes.io/component": "shenyu-bootstrap"
    "app.kubernetes.io/instance": "shenyu-dev"
  type: NodePort
  ports:
    - protocol: TCP
      port: 9195
      targetPort: 9195
      nodePort: 31195
---
# Source: shenyu/templates/shenyu-admin-deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  namespace: shenyu
  name: shenyu-dev-admin
  labels:
    "app.kubernetes.io/name": 'shenyu-admin'
    "app.kubernetes.io/managed-by": "Helm"
    "app.kubernetes.io/component": "shenyu-admin"
    "app.kubernetes.io/instance": "shenyu-dev"
spec:
  selector:
    matchLabels:
      "app.kubernetes.io/component": "shenyu-admin"
      "app.kubernetes.io/instance": "shenyu-dev"
  replicas: 1
  template:
    metadata:
      labels:
        "app.kubernetes.io/name": 'shenyu-admin'
        "app.kubernetes.io/managed-by": "Helm"
        "app.kubernetes.io/component": "shenyu-admin"
        "app.kubernetes.io/instance": "shenyu-dev"
    spec:
      containers:
        - name: shenyu-admin
          image: apache/shenyu-admin:2.4.3
          imagePullPolicy: Always
          ports:
            - containerPort: 9095
          env:
            - name: 'TZ'
              value: 'Asia/Beijing'
          volumeMounts:
          - name: shenyu-admin-config
            mountPath: /opt/shenyu-admin/config/application-h2.yml
            subPath: application-h2.yml
      volumes:
        - name: shenyu-admin-config
          configMap:
            name: shenyu-dev-configmap
            items:
            - key: application-h2.yml
              path: application-h2.yml
---
# Source: shenyu/templates/shenyu-bootstrap-deployment.yaml
# shenyu-bootstrap
apiVersion: apps/v1
kind: Deployment
metadata:
  namespace: shenyu
  name: shenyu-dev-bootstrap
  labels:
    "app.kubernetes.io/name": 'shenyu-bootstrap'
    "app.kubernetes.io/managed-by": "Helm"
    "app.kubernetes.io/component": "shenyu-bootstrap"
    "app.kubernetes.io/instance": "shenyu-dev"
spec:
  selector:
    matchLabels:
      "app.kubernetes.io/component": "shenyu-bootstrap"
      "app.kubernetes.io/instance": "shenyu-dev"
  replicas: 1
  template:
    metadata:
      labels:
        "app.kubernetes.io/name": 'shenyu-bootstrap'
        "app.kubernetes.io/managed-by": "Helm"
        "app.kubernetes.io/component": "shenyu-bootstrap"
        "app.kubernetes.io/instance": "shenyu-dev"
    spec:
      volumes:
        - name: shenyu-bootstrap-config
          configMap:
            name: shenyu-dev-configmap
            items:
              - key: application-local.yml
                path: application-local.yml
      containers:
        - name: shenyu-bootstrap
          image: apache/shenyu-bootstrap:2.4.3
          ports:
            - containerPort: 9195
          env:
            - name: TZ
              value: Asia/Beijing
          volumeMounts:
            - name: shenyu-bootstrap-config
              mountPath: /opt/shenyu-bootstrap/conf/application-local.yml
              subPath: application-local.yml
```

### MySQL

**Command:** 
```shell
helm install shenyu shenyu/shenyu -n=shenyu --create-namespace --debug --dry-run \
      --set dataSource.active=mysql \
      --set dataSource.mysql.ip=127.0.0.1 \
      --set dataSource.mysql.port=3306 \
      --set dataSource.mysql.password=123456 \
      --set dataSource.mysql.storageClass=shenyu-local-storage
```

Output:
```yaml
install.go:178: [debug] Original chart version: ""
install.go:195: [debug] CHART PATH: /Users/lhp/Library/Caches/helm/repository/shenyu-2.4.4.tgz

NAME: shenyu
LAST DEPLOYED: Wed Aug 17 19:56:51 2022
NAMESPACE: shenyu
STATUS: pending-install
REVISION: 1
TEST SUITE: None
USER-SUPPLIED VALUES:
dataSource:
  active: mysql
  mysql:
    ip: 127.0.0.1
    password: 123456
    port: 3306
    storageClass: shenyu-local-storage

COMPUTED VALUES:
admin:
  enabled: true
  image: apache/shenyu-admin
  nodePort: 31095
bootstrap:
  enabled: true
  image: apache/shenyu-bootstrap
  nodePort: 31195
dataSource:
  active: mysql
  h2:
    enabled: true
  mysql:
    enabled: false
    ip: 127.0.0.1
    password: 123456
    port: 3306
    storageClass: shenyu-local-storage
    username: root
replicas: 1
version: 2.4.3

HOOKS:
MANIFEST:
---
# Source: shenyu/templates/ConfigMap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  namespace: shenyu
  name: shenyu-configmap
  labels:
    "app.kubernetes.io/name": 'shenyu'
    "app.kubernetes.io/managed-by": "Helm"
    "app.kubernetes.io/component": "shenyu-configmap"
    "app.kubernetes.io/instance": "shenyu"
data:
  application-local.yml: |
    server:
        port: 9195
        address: 0.0.0.0
    spring:
        main:
            allow-bean-definition-overriding: true
        application:
            name: shenyu-bootstrap
    management:
        health:
            defaults:
            enabled: false
    shenyu:
        local:
            enabled: true
        file:
            enabled: true
        cross:
            enabled: true
        dubbo:
            parameter: multi
        sync:
            websocket:
              urls: ws://shenyu-admin.shenyu.svc.cluster.local:9095/websocket
        exclude:
            enabled: false
            paths:
            - /favicon.ico
        extPlugin:
            enabled: true
            threads: 1
            scheduleTime: 300
            scheduleDelay: 30
        scheduler:
            enabled: false
            type: fixed
            threads: 16
    logging:
        level:
            root: info
            org.springframework.boot: info
            org.apache.ibatis: info
            org.apache.shenyu.bonuspoint: info
            org.apache.shenyu.lottery: info
            org.apache.shenyu: info
---
# Source: shenyu/templates/shenyu-admin-svc.yaml
# Now we only support NodePort mode
apiVersion: v1
kind: Service
metadata:
  namespace: shenyu
  name: shenyu-admin
  labels:
    "app.kubernetes.io/name": 'shenyu-admin'
    "app.kubernetes.io/managed-by": "Helm"
    "app.kubernetes.io/component": "shenyu-admin"
    "app.kubernetes.io/instance": "shenyu"
spec:
  selector:
    "app.kubernetes.io/component": "shenyu-admin"
    "app.kubernetes.io/instance": "shenyu"
  type: NodePort
  ports:
    - protocol: TCP
      port: 9095
      targetPort: 9095
      nodePort: 31095
---
# Source: shenyu/templates/shenyu-bootstrap-svc.yaml
# Now we only support NodePort mode
apiVersion: v1
kind: Service
metadata:
  namespace: shenyu
  name: shenyu-bootstrap
  labels:
    "app.kubernetes.io/name": 'shenyu-bootstrap'
    "app.kubernetes.io/managed-by": "Helm"
    "app.kubernetes.io/component": "shenyu-bootstrap"
    "app.kubernetes.io/instance": "shenyu"
spec:
  selector:
    "app.kubernetes.io/component": "shenyu-bootstrap"
    "app.kubernetes.io/instance": "shenyu"
  type: NodePort
  ports:
    - protocol: TCP
      port: 9195
      targetPort: 9195
      nodePort: 31195
---
# Source: shenyu/templates/shenyu-admin-deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  namespace: shenyu
  name: shenyu-admin
  labels:
    "app.kubernetes.io/name": 'shenyu-admin'
    "app.kubernetes.io/managed-by": "Helm"
    "app.kubernetes.io/component": "shenyu-admin"
    "app.kubernetes.io/instance": "shenyu"
spec:
  selector:
    matchLabels:
      "app.kubernetes.io/component": "shenyu-admin"
      "app.kubernetes.io/instance": "shenyu"
  replicas: 1
  template:
    metadata:
      labels:
        "app.kubernetes.io/name": 'shenyu-admin'
        "app.kubernetes.io/managed-by": "Helm"
        "app.kubernetes.io/component": "shenyu-admin"
        "app.kubernetes.io/instance": "shenyu"
    spec:
      containers:
        - name: shenyu-admin
          image: apache/shenyu-admin:2.4.3
          imagePullPolicy: Always
          ports:
            - containerPort: 9095
          env:
            - name: 'TZ'
              value: 'Asia/Beijing'
---
# Source: shenyu/templates/shenyu-bootstrap-deployment.yaml
# shenyu-bootstrap
apiVersion: apps/v1
kind: Deployment
metadata:
  namespace: shenyu
  name: shenyu-bootstrap
  labels:
    "app.kubernetes.io/name": 'shenyu-bootstrap'
    "app.kubernetes.io/managed-by": "Helm"
    "app.kubernetes.io/component": "shenyu-bootstrap"
    "app.kubernetes.io/instance": "shenyu"
spec:
  selector:
    matchLabels:
      "app.kubernetes.io/component": "shenyu-bootstrap"
      "app.kubernetes.io/instance": "shenyu"
  replicas: 1
  template:
    metadata:
      labels:
        "app.kubernetes.io/name": 'shenyu-bootstrap'
        "app.kubernetes.io/managed-by": "Helm"
        "app.kubernetes.io/component": "shenyu-bootstrap"
        "app.kubernetes.io/instance": "shenyu"
    spec:
      volumes:
        - name: shenyu-bootstrap-config
          configMap:
            name: shenyu-configmap
            items:
              - key: application-local.yml
                path: application-local.yml
      containers:
        - name: shenyu-bootstrap
          image: apache/shenyu-bootstrap:2.4.3
          ports:
            - containerPort: 9195
          env:
            - name: TZ
              value: Asia/Beijing
          volumeMounts:
            - name: shenyu-bootstrap-config
              mountPath: /opt/shenyu-bootstrap/conf/application-local.yml
              subPath: application-local.yml

```

### 1. Install with h2 on Mac

image

image

### 2. Install with MySQL on Linux

image

image
image

image

image

image

image

### CODE REVIEW

**Comment by erdengk** [comment]
@aFlyBird0 
I tested the h2 data source and it works fine.
I suggest dividing this pr into two prs, one supports h2 and one supports MySQL, what do you think?

**Comment by aFlyBird0** [comment]
> @aFlyBird0 I tested the h2 data source and it works fine. I suggest dividing this pr into two prs, one supports h2 and one supports MySQL, what do you think?

Thank you for the advice! I think it's in the beginning stage of the project and the changes are more significant. So I would like to finish testing MySQL as soon as possible, and the documentation is perfected, and then ask for testing and review. @erdengk 

When the whole project works, we can split the pr as small as possible.

**Comment by aFlyBird0** [comment]
Please do not merge this pr directly, as the GitHub Action yaml file is now set to a test state by me. After approving it, please @ me and I will modify it.

**Comment by aFlyBird0** [comment]
@erdengk @JooKS-me Ready for review, thank you ❤️

File: charts/shenyu/README.md [hunk 1]
```diff
@@ -1,3 +1,138 @@
...
+
+#### 2. 安装
+
+修改以下命令并复制，执行：
+
+其中，storageClass 和上面的 yaml 创建的 StorageClass 的 name 对应。
+
+```shell
+helm install shenyu shenyu/shenyu -n=shenyu --create-namespace \
+      --set dataSource.mysql.enabled=false \
```
Comment (JooKS-me) [inline charts/shenyu/README.md]: Remove this line

**Review by JooKS-me — CHANGES_REQUESTED** [review]

File: charts/shenyu/README.md (hunk 1 above)
Comment (aFlyBird0) [reply charts/shenyu/README.md]: done. Now you can just use `dataSource.active.mysql` to enable mysql.

**Review by JooKS-me — APPROVED** [review]

**Pull request merged**